sdk/tests/types/unlock/alias.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_sdk::types::block::{unlock::AliasUnlock, Error};
+use packable::{bounded::InvalidBoundedU16, PackableExt};
+
+#[test]
+fn kind() {
+    assert_eq!(AliasUnlock::KIND, 2);
+}
+
+#[test]
+fn new_valid_min_index() {
+    assert_eq!(AliasUnlock::new(0).unwrap().index(), 0);
+}
+
+#[test]
+fn new_valid_max_index() {
+    assert_eq!(AliasUnlock::new(126).unwrap().index(), 126);
+}
+
+#[test]
+fn new_invalid_more_than_max_index() {
+    assert!(matches!(
+        AliasUnlock::new(128),
+        Err(Error::InvalidAliasIndex(InvalidBoundedU16(128)))
```

Comment:
Use 127 already?
```suggestion
        AliasUnlock::new(127),
        Err(Error::InvalidAliasIndex(InvalidBoundedU16(127)))
```

Reply:
Actually the UNLOCK_INDEX_RANGE is wrong?

Reply:
Hmm no ? 127 is valid, 128 is not?

Reply:
How can 127 be valid if 127 is the highest possible unlock? It would be a self reference

Reply:
127 is allowed by the TIP, but I agree with you, it would fail during validation of `Unlocks`, but not here 🤷🏻

Reply:
Then the TIP should also get updated? :carlos:

Reply:
Maybe yeah, it's just a bit weird but in the end it's all validated correctly

Reply:
Opened a PR to update the TIP https://github.com/iotaledger/tips/pull/134

Reply:
Can we still merde this until they accept it? This is going to break more tests anyway so I'd do another PR